PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs

```diff
@@ -92,17 +101,81 @@ public DoctorAppointment RecommendAnAppointment(DoctorUser doctor, DateTime date
             return null;
         }
 
+        private List<DoctorAppointment> GetAllAppointmentsByDateAndDoctor(DateTime date, int doctorId)
+        {
+            List<DoctorAppointment> appointments = new List<DoctorAppointment>();
+            foreach (DoctorAppointment appointment in GetAppointmentsForDoctor(doctorId))
+            {
+                if(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date))
+                {
+                    appointments.Add(appointment);
+                }
+            }
+            return appointments;
```

Comment:
Refactor code. Replace method etAllAppointmentsByDateAndDoctor(DateTime date, int doctorId) with Where syntax, for example :
return GetAppointmentsForDoctor(doctorId).Where(appointment  =>(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date)) );